jax/tools/jax_to_ir.py

```diff
@@ -241,6 +241,7 @@ def parse_shape_str(s):
 _DT = {'pred': jnp.bool_,
        'u8': jnp.uint8, 'u16': jnp.uint16, 'u32': jnp.uint32, 'u64': jnp.uint64,
        's8': jnp.int8, 's16': jnp.int16, 's32': jnp.int32, 's64': jnp.int64,
+       'f8e4m3fnuz': jnp.float8_e4m3fnuz, 'f8e5m2fnuz': jnp.float8_e5m2fnuz,
```

Comment:
Why do we include these types here but not other float8 types? Should there be a TODO?

Reply:
I've scoped this PR to only the `fnuz` types. This is just a matter of taste. It makes it so any potential future reverts don't lose unrelated functionality, for example.

I'm happy to add the other types, If you prefer.

Reply:
In other words, why is this required for the new fp8 types, but not for the old ones? Are there tests that are exercising this code for the new types but not for the existing ones?

Reply:
> In other words, why is this required for the new fp8 types

To support `jax_to_ir` tool with these fp8 types.
> but not for the old ones

I don't think this PR is saying they aren't required, it just isn't making that change as per my previous comment.
> Are there tests that are exercising this code for the new types but not for the existing ones

Yes, in `jax_to_ir_tests.py`.

Reply:
You mean the tests added in this PR? OK, I see.

I guess this is just more for my own information: I don't know what this part of the code does, and I'm wondering what functionality is unlocked by including these types here, and why it hasn't been a problem thus far that none of the existing float8 types are included here. Why has nobody complained about this being missing before?

Reply:
I've dropped these changes. 
> Why has nobody complained about this being missing before?

I don't really have an answer to this. One answer might be FP8 hardware support is still pretty uncommon. Another might be nobody uses this tool. I've added these fp8 types where they _seem_ relevant. Perhaps this tool should be dropped altogether if it has no users.